src/main/scala/org/zalando/spearheads/innkeeper/dao/PathsPostgresRepo.scala

```diff
@@ -73,15 +73,37 @@ class PathsPostgresRepo @Inject() (
     }
   }
 
-  override def pathWithUriHostIdExists(uri: String, hostIds: Seq[Long]): Future[Boolean] = {
-    logger.debug("uri hostId pairs are duplicate check")
+  override def collisionExistsForPath(path: PathIn): Future[Boolean] = {
+    logger.debug("path uri collision check")
+
+    def position = SimpleExpression.binary[String, String, Int] {
+      (subString, string, queryBuilder) =>
+        import slick.util.MacroSupport._
```

Comment:
Those are specific imports for this particular function, one of which imports from a parameter. Having these imports here makes it so that when reading the function you know exactly why there is some magic happening there.